src/components/Calculator.js

```diff
@@ -1,34 +1,48 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = { operand: {} };
+  }
+
+  printOn = (event) => {
+    const buttonName = event.target.textContent;
+    const { operand } = this.state;
+    this.setState({
+      operand: calculate(operand, buttonName),
+    });
   }
 
   render() {
+    const { operand } = this.state;
     return (
       <div className="calculator-grid">
-        <div className="output">0</div>
-        <button type="button">AC</button>
-        <button type="button">+/-</button>
-        <button type="button">%</button>
-        <button type="button" className="operator">÷</button>
-        <button type="button">7</button>
-        <button type="button">8</button>
-        <button type="button">9</button>
-        <button type="button" className="operator">x</button>
-        <button type="button">4</button>
-        <button type="button">5</button>
-        <button type="button">6</button>
-        <button type="button" className="operator">-</button>
-        <button type="button">1</button>
-        <button type="button">2</button>
-        <button type="button">3</button>
-        <button type="button" className="operator">+</button>
-        <button type="button" className="occupy-two">0</button>
-        <button type="button">.</button>
-        <button type="button" className="operator">=</button>
+        <div className="output">
+          {operand.next}
+          {operand.operation}
+          {operand.total}
+        </div>
```

Comment:
[OPTIONAL] I think it would be good to show the classic 0 when there's no operation currently running on the calculator instead of a blank screen. This is kind of a standard on all calculators that tells the user the calc is ready to accept inputs. What do you think?

Reply:
Absolutely! I will definitely look into implementing that, thanks a lot for your comment!